lib/firestore/models.dart

```diff
@@ -151,19 +150,23 @@ class CollectionReference extends Reference {
 
   void _addToComposite(StructuredQuery_Filter filter) {
     StructuredQuery_CompositeFilter compositeFilter;
-    if (_structuredQuery.hasWhere() &&
-        _structuredQuery.where.hasCompositeFilter()) {
-      compositeFilter = _structuredQuery.where.compositeFilter;
+    if (_getStructuredQuery().hasWhere() &&
+        _getStructuredQuery().where.hasCompositeFilter()) {
+      compositeFilter = _getStructuredQuery().where.compositeFilter;
     } else {
       compositeFilter = StructuredQuery_CompositeFilter()
         ..op = StructuredQuery_CompositeFilter_Operator.AND;
     }
 
     compositeFilter.filters.add(filter);
-    _structuredQuery.where = StructuredQuery_Filter()
+    _getStructuredQuery().where = StructuredQuery_Filter()
       ..compositeFilter = compositeFilter;
   }
 
+  StructuredQuery _getStructuredQuery() =>
+      _structuredQuery ??= StructuredQuery()
+        ..from.add(StructuredQuery_CollectionSelector()..collectionId = id);
```

Comment:
It'll be hard to guarantee that this getter will always be accessed rather than directly accessing the field in the future.
It also seems like the collection id is being added every time _structuredQuery is accessed. Is that safe?

In any case I tested here with your old code and checking for initialisation rather than nullability and it seems to work fine:

``` dart
Future<Page<Document>> get(
          {int pageSize = 1024, String nextPageToken = ''}) =>
      _structuredQuery.isInitialized()
          ? _gateway.getCollection(...)
          : gateway.runQuery(...);
```

Do you think that would be a better option?

Reply:
??= means that the value is only changed if the object is null

Reply:
You're right, I'm sometimes confused by the cascading operator. I guess it doesn't cascade if the null condition isn't true.